playwright/masonry/scroll-performance.spec.mjs

```diff
@@ -0,0 +1,96 @@
+// @flow strict
+import fs from 'fs';
+import { expect, test } from '@playwright/test';
+import getServerURL from './utils/getServerURL.mjs';
+import {
+  eventsToCsv,
+  getFpsFromEvents,
+  getFpsMetricsPerScroll,
+  metricsToCsv,
+} from './utils/tracingEvents.mjs';
+import waitForRenderedItems from './utils/waitForRenderedItems.mjs';
+
+/*::
+type Event = {
+  name: string,
+  ts: number,
+  fps?: number,
+};
+*/
+
+const DEFAULT_SCROLL_COUNT = 5;
+const PINS_COUNT = 20;
+const PAGE_HEIGHT = 1000;
+
+test.describe('Masonry: scrolls', () => {
+  test('scroll test', async ({ page }) => {
+    const events /*: Array<Event> */ = [];
+
+    // Starting the CDP session only including the desired categories
+    const client = await page.context().newCDPSession(page);
+    await client.send('Tracing.start', {
+      traceConfig: {
+        includedCategories: ['benchmark', 'clock_sync'],
+      },
+    });
+
+    // This will be called when tracing is complete
+    client.on('Tracing.dataCollected', (data) => {
+      events.push(
+        ...data.value
+          .filter((e) =>
+            ['DirectRenderer::DrawFrame', 'clock_sync'].includes(e.name)
+          )
+          .map((e) => ({ name: e.name, ts: e.ts }))
+      );
+    });
+
+    const tracingCompleteEvent = new Promise((resolve) => {
+      client.on('Tracing.tracingComplete', () => {
+        resolve();
+      });
+    });
+
+    await page.setViewportSize({ width: 1600, height: PAGE_HEIGHT });
+    await page.goto(getServerURL());
```

Comment:
not sure if it makes a difference but should we also add a `waitForRenderedItems` here to wait for the initial render.

also, just noticed that based on where this is currently placed we likely are also capturing `DrawFrame` events from the initial render which could be skewing the actual FPS calculations